src/Form.js

```diff
@@ -0,0 +1,221 @@
+import React, { Component } from 'react'
+import './App.css'
+import axios from 'axios'
+
+class Form extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      name: '',
+      email: '',
+      pass: '',
+      avatar: '',
+      phone: '',
+      age: 1,
+      gender: 'male',
+      favorite: {
+        soccer: false,
+        badminton: false,
+        sailing: false
+      },
+      note: "Buổi học đầu tiên về html"
+    }
+  }
+
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  handleChangeFavorite = event => {
+    const check = event.target.checked
+    const value = event.target.value
+    let favorites = this.state.favorite
+    for (const key in favorites) {
+      if (favorites.hasOwnProperty(key)) {
+        if(key === value) {
+          favorites[key] = check
+        }
+      }
+    }
+    this.setState({
+      favorite: favorites
+    });
+  }
+
+  handleSubmit = event => {
+    event.preventDefault()
+    axios.post(`https://jsonplaceholder.typicode.com/users`, this.state)
```

Comment:
không gửi toàn bộ state lên API, có thể trong `state` có data thừa, trong trường hợp này có thể không có data thừa, nhưng hãy chú ý về sau
Chỉ lấy những gì cần thiết thì mới gửi lên thôi